explorer/interpreter/interpreter.cpp

```diff
@@ -860,6 +865,42 @@ auto Interpreter::Convert(Nonnull<const Value*> value,
       }
       return Convert(value, destination_type, source_loc);
     }
+    case Value::Kind::PointerValue: {
+      if (destination_type->kind() != Value::Kind::PointerType ||
```

Comment:
I'm trying to understand this. The value here is some memory address. The destination type is, say, i32. The conversion result is the memory address, i.e. success?

I think this needs to be tested a bit more; do we have things like:

```
var a: D = {};
var b: D* = &a:
var c: i32 = b;
```

```
var a: i32 = 0;
var b: i32* = &a;
var c: D* = b;
```

```
var a: A = 0;
var b: A* = &a;
var c: A** = b;
```

```
var a: A = 0;
var b: A* = &a;
var c: A* = b;
```

Reply:
So, I think there are two possibilities here:
1) The code path is unused, CARBON_CHECK is okay
2) The code started running

I think the suggested tests currently fail, e.g.:
https://carbon.compiler-explorer.com/z/s9ecvezfe
https://carbon.compiler-explorer.com/z/qe64jEbYr

So let's add the tests explicitly, and make sure there's no regression.

Reply:
Tests added and passing ✔️